java/ql/src/Security/CWE/CWE-079/AndroidWebViewAddJavascriptInterface.ql

```diff
@@ -0,0 +1,17 @@
+/**
+ * @id java/android-webview-addjavascriptinterface
+ * @description Exposing a Javascript interface to a Java object in a WebView can lead to malicious JavaScript controlling the application.
+ * @kind problem
+ * @problem.severity warning
+ * @security-severity 6.1
+ * @precision high
+ * @tags security
+ *       external/cwe/cwe-079
+ */
+
+import java
+import semmle.code.java.frameworks.android.WebView
+
+from MethodAccess ma
+where ma.getMethod() instanceof WebViewAddJavascriptInterfaceMethod
```

Comment:
I think we could make this query slightly smarter by discarding methods that check the URL origin, see https://github.com/OWASP/owasp-mastg/blob/28116d4ac1b1187893687f416ffc5d0fc6898bd9/Document/0x05h-Testing-Platform-Interaction.md#determining-whether-java-objects-are-exposed-through-webviews-mstg-platform-7.

> If `addJavascriptInterface` is necessary, take the following considerations:
> * Only JavaScript provided with the APK should be allowed to use the bridges, e.g. by verifying the URL on each bridged Java method (via WebView.getUrl).